core-common/src/main/java/org/mapstruct/Mapper.java

```diff
@@ -85,6 +85,28 @@
     String componentModel() default "default";
 
     /**
+     * Specifies the name of the implementation class. A single {@code *} wildcard can be used and will be replaced
+     * by the interface's or abstract class' name.
+     * <p>
+     * Defaults to postfixing the name with {@code Impl}
+     *
+     * @return The implementation suffix.
+     * @see #implementationPackage()
+     */
+    String implementationName() default "default";
```

Comment:
> I wish we could use "<CLASS_NAME>Impl" as the annotation default value (because that's what will be used), but I don't think we can because otherwise we could not distinguish the case where a specific @Mapper annotation overrides the default from @MapperConfig from the case where the inherited default should be used.

I thought about this the other day while driving on a four hour trip.... The thing is, actually we could distinguish it - but as we're using hickory to generate the annotation-prisms, we're used to rely on those prisms to access the annotation values. The APT API would give us the information if a value was explicitly specified or not - the hickory-generated prims just return the default values in those cases, for convinience.
I'll create a speparate issue where we can discuss/find a better solution and with that we can then clean up all our annotation definitions.